app/models/concerns/profile_section.rb

```diff
@@ -5,8 +5,9 @@ module ProfileSection
     def prepare(**, &block)
       find_or_initialize_by(**).tap do |record|
         if record.new_record?
-          if (previous_application = jobseeker(record).job_applications.last)
-            copy_attributes(record, previous_application)
+          # do not populated from draft applications as they make be incomplete and invalid.
+          if (previously_submitted_application = jobseeker(record).job_applications.after_submission.last)
```

Comment:
`after_submission` didn't know we had this. Nice!

Reply:
me neither, @starswan pointed it out in an earlier comment!